server/controllers/availability.js

```diff
@@ -0,0 +1,21 @@
+const asyncHandler = require("express-async-handler");
+const mongoose = require("mongoose");
+const User = require("../models/User");
+const Profile = require("../models/Profile");
+
+exports.updateAvailability = asyncHandler(async (req, res, next) => {
+  const { id } = req.user;
+  if (!id) {
+    res.status(400).json({ error });
+  }
```

Comment:
Good validation, but it seems `error` is not defined within this scope

Reply:
@ezeilo-su I was wondering since i am using protect function(which checks if user is logged) in the route this is kind of redundant code. isnt it